File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/utils/RabbitUtils.java
package org.xyattic.eventual.consistency.support.core.utils;

import lombok.SneakyThrows;

/**
 * @author wangxing
 * @create 2020/3/28
 */
public class RabbitUtils {

    @SneakyThrows
    private static String get(String str) {
//        List<String> activeProfiles =
//                Arrays.asList(SpringContextUtils.getEnvironment().getActiveProfiles());
//        return activeProfiles.contains("local") ||
//                activeProfiles.contains("dev") ? WebUtils.getServerIpAddress().getHostAddress() + "_" + str : str;
        return str;
    }

    public static String getQueue(String queue) {
        return get(queue);
    }

    public static String getRoutingKey(String routingKey) {
        return get(routingKey);
    }

    public static String getExchange(String exchange) {
        return get(exchange);
    }

}

File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/utils/RabbitUtils.kt
package org.xyattic.eventual.consistency.support.core.utils

/**
 * @author wangxing
 * @create 2020/3/28
 */
class RabbitUtils {

    companion object {

        @JvmStatic
        fun getQueue(queue: String): String {
            return RabbitUtils.get(queue)
        }

        @JvmStatic
        fun getRoutingKey(routingKey: String): String {
            return RabbitUtils.get(routingKey)
        }

        @JvmStatic
        fun getExchange(exchange: String): String {
            return RabbitUtils.get(exchange)
        }

        private fun get(str: String): String {
//        List<String> activeProfiles =
//                Arrays.asList(SpringContextUtils.getEnvironment().getActiveProfiles());
//        return activeProfiles.contains("local") ||
//                activeProfiles.contains("dev") ? WebUtils.getServerIpAddress().getHostAddress() + "_" + str : str;
            return str
        }
    }


}